app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -1,29 +1,65 @@
 package com.lightricks.feedexercise.data
 
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Observer
+import androidx.room.Room
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.google.common.truth.Truth
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.network.MockFeedApiService
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
 import org.junit.runner.RunWith
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
 @RunWith(AndroidJUnit4::class)
 class FeedRepositoryTest {
-   //todo: add the tests here
+
+    private lateinit var dBase: FeedDatabase
+    private lateinit var repository: FeedRepository
+
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+    @Before
+    fun initDb() {
+        dBase = Room.inMemoryDatabaseBuilder(
+            InstrumentationRegistry.getInstrumentation().context,
+            FeedDatabase::class.java
+        ).build()
+        repository = FeedRepository(dBase, MockFeedApiService())
+    }
+
+    @After
+    fun destroyDb() {
+        dBase.close()
+    }
+
+    @Test
+    fun refresh() {
+        repository.refresh().test().await()
```

Comment:
Can you split this into 2 tests (one for saving and another one for loading)?
This way we will know exactly which step fails.

Reply:
Done on 1.1